google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionImpl.java

```diff
@@ -98,12 +99,14 @@ interface SessionTransaction {
   ByteString readyTransactionId;
   private final Map<SpannerRpc.Option, ?> options;
   private Span currentSpan;
+  private volatile Instant lastUseTime;
 
   SessionImpl(SpannerImpl spanner, String name, Map<SpannerRpc.Option, ?> options) {
     this.spanner = spanner;
     this.options = options;
     this.name = checkNotNull(name);
     this.databaseId = SessionId.of(name).getDatabaseId();
+    this.lastUseTime = Instant.now();
```

Comment:
Should we use the clock object here as well for uniformity?

Reply:
* Clock object is only added so that it allows tests to mock a particular class. In this case, `SessionImpl` is never mocked given it is a `stateful object` . It contains a continuosly changing value `lastUseTime`. Hence, at all tests we are now using a real instance of `SessionImpl`. 
* Given we don't have an use-case to mock the `SessionImpl` object anywhere, we don't need the clock object here.